lib/iex/lib/iex/introspection.ex

```diff
@@ -115,13 +126,37 @@ defmodule IEx.Introspection do
         end
         :ok
       else
-        :not_found
+        if has_callback?(mod, fun, arity) do
+          :behaviour_found
+        else
+          :not_found
+        end
       end
     else
       :no_docs
     end
   end
 
+  defp has_callback?(mod, fun) do
+    mod
+    |> Code.get_docs(:callback_docs)
+    |> find_callback_doc(fun)
+  end
+
+  defp has_callback?(mod, fun, arity) do
+    mod
+    |> Code.get_docs(:callback_docs)
+    |> find_callback_doc(fun, arity)
+  end
+
+  def find_callback_doc(docs, fun) do
+    Enum.any?(docs, &match?({{^fun, _}, _, _, _}, &1))
+  end
+
+  def find_callback_doc(docs, fun, arity) do
```

Comment:
Ups, missed the `p` when cleaning it up...
I personally think it does make understanding the goal of the code easier, that's why I moved it to a separate function.

If other team members agree on not extracting it, it's fine to me. Meanwhile I'll fix the `defp` issue.

Thanks!